### PR TITLE
Fix #6381

### PR DESCRIPTION
**Pull request by yanzhiwei147**
Fix [#6381](https://github.com/CocoaPods/CocoaPods/issues/6381)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,7 +22,9 @@
 
 ##### Bug Fixes
 
-* None.
+* Fix TypeError - no implicit conversion of nil into String.
```
Comment (segiddins) [inline CHANGELOG.md]: this should say what the underlying bug is, rather than the exception being avoided

File: lib/cocoapods-core/source.rb [hunk 2]
```diff
@@ -160,7 +160,7 @@ def versions(name)
       return nil unless specs_dir
       raise ArgumentError, 'No name' unless name
       pod_dir = pod_path(name)
-      return unless pod_dir.exist?
+      return nil unless pod_dir.exist?
```
Comment (segiddins) [inline lib/cocoapods-core/source.rb]: this is unnecessary -- `return` will implicitly return `nil`

File: lib/cocoapods-core/source/aggregate.rb [hunk 3]
```diff
@@ -51,6 +51,7 @@ def all_sets
       #         name. Returns nil if no representative source found containing a pod with given name.
       #
       def representative_set(name)
+        warn '[DEPRECATION] `representative_set` is deprecated.  Please use `representative_set_for_search` instead.'
```
Comment (segiddins) [inline lib/cocoapods-core/source/aggregate.rb]: please mark as deprecated in the API docs as well

File: lib/cocoapods-core/source.rb (hunk 2 above)
Comment (yanzhiwei147) [reply lib/cocoapods-core/source.rb]: OK,but why the line 161 write `return nil unless specs_dir`?

File: lib/cocoapods-core/source/aggregate.rb (hunk 3 above)
Comment (yanzhiwei147) [reply lib/cocoapods-core/source/aggregate.rb]: OK

**Comment by yanzhiwei147** [comment]
@dnkoutso This PR is new for https://github.com/CocoaPods/Core/pull/368

**Comment by dnkoutso** [comment]
@yanzhiwei147 please rebase and squash ino a single commit.

**Comment by yanzhiwei147** [comment]
[New PR](https://github.com/CocoaPods/Core/pull/405) provider, close this.

**Pull request closed without merging**